src/ROOTWriter.cc

```diff
@@ -55,7 +55,18 @@ void ROOTWriter::createBranches(const std::vector<StoreCollection>& collections)
     const auto collBuffers = coll->getBuffers();
     if (collBuffers.data) {
       // only create the data buffer branch if necessary
-      const auto collClassName = "vector<" + coll->getValueTypeName() + "Data>";
+
+      std::string valTypeName = coll->getValueTypeName() ;
+
+      auto collClassName = "vector<" + valTypeName + "Data>";
+
+      auto it = valTypeName.find("podio::User_") ;
+      if(it == 0 ){
+	valTypeName = valTypeName.substr( 12 , 1024 ) ; // rm 'podio_user_' prefix
+	 collClassName = "vector<" + valTypeName +">";
+      }
```

Comment:
Should we consider adding a `getDataTypeName` function (or similar) to the `podio::CollectionBase`? It would make some of the string juggling easier to do for general use, and give us a customization point where these user classes could be handled more nicely. As far as I can tell the major use cases for `getValueTypeName` currently are:
- Keys for the map inside the `SIOBlockFactory`
- Creating the `vector<XYZData>` for the root writer and reader

So maybe it could even be a `getDataVectorTypeName` alongside the `getValueTypeName` (?)